force-app/main/default/lwc/serviceDeliveryRow/serviceDeliveryRow.js

```diff
@@ -118,264 +88,197 @@ export default class ServiceDeliveryRow extends LightningElement {
-        ) {
-            this.handleProgramEngagementInputChange(event);
-        } else if (event.target.fieldName === this.fields.service.fieldApiName) {
-            this.handleServiceInputChange(event.target.fieldName, event.target.value);
+    @api
+    saveRow() {
+        if (!this.isDirty) {
+            return;
+        }
+        let deliverySubmit = this.template.querySelector(".sd-submit");
+        if (deliverySubmit) {
+            deliverySubmit.click();
         }
     }
 
-    handleServiceInputChange(fieldName, fieldVal) {
-        this.serviceId = fieldVal;
-        this.handleEnableFieldOnInputChange(fieldName);
-        this.enableDisableFieldsOnSaveAndInputChange();
+    @api
+    get defaultValues() {
+        return this._defaultValues;
     }
-
-    handleContactInputChange(event) {
-        if (this.hasContactField && this.hasProgramEngagementField) {
-            if (event.detail.value && event.detail.value.length) {
-                this.selectedContact = event.detail.value[0];
-                this.handleGetServicesEngagements(event.detail.value[0]);
-            } else {
-                this.handleResetContact();
-            }
-        } else if (this.hasContactField && !this.hasProgramEngagementField) {
-            this.handleEnableFieldOnInputChange(event.target.fieldName);
-        }
+    set defaultValues(value) {
+        this._defaultValues = value;
+        this.setDefaults();
     }
 
-    handleProgramEngagementInputChange(event) {
-        if (event.detail.value && event.detail.value.length) {
-            this.isServiceFiltered = true;
-            this.handleGetServicesForProgramEngagement(event.detail.value[0]);
-        }
+    @api
+    get serviceDeliveryFieldSets() {
+        return this._fieldSets;
     }
+    set serviceDeliveryFieldSets(value) {
+        if (!value) {
+            return;
+        }
 
-    handleEnableFieldOnInputChange(fieldApiName) {
-        //Getting this error Uncaught TypeError: 'set' on proxy: when trying to enable an element
-        //on Input field change and we suspect that since the record edit form is updating the values on the same array and
-        //that is the reason why we are cloning the object here
-        this.localFieldSet = JSON.parse(JSON.stringify(this.localFieldSet));
-        this.localFieldSet.forEach(element => {
-            if (fieldApiName !== element.apiName) {
-                element.disabled = false;
-            }
-        });
+        this._fieldSets = value;
+        this._fieldSets.currentFieldSetName = value.currentFieldSetName
+            ? value.currentFieldSetName
+            : DEFAULT_FIELD_SET;
+
+        this.setCurrentFieldSet();
     }
 
-    handleResetContact() {
-        this.localFieldSet.forEach(element => {
-            element.showFilteredInput = false;
-        });
-        this._filteredValues = [];
-        this._valuesToSave = [];
+    get isDeleteDisabled() {
```

Comment:
I was on the fence for this.  Being able to delete the only row visible would be a quick way to reset it.  Nothing to worry about now, but something to consider.